cmake/sysbuild/image_signing.cmake

```diff
@@ -57,7 +57,14 @@ function(zephyr_mcuboot_tasks)
   # back on mcuboot/scripts/imgtool.py. We exclude the system imgtool when
   # compressed image support is enabled due to needing a version of imgtool
   # that has features not in the most recent public release.
-  if(IMGTOOL AND NOT CONFIG_MCUBOOT_COMPRESSED_IMAGE_SUPPORT_ENABLED)
+  # Similar "pure" siganture and SHA512 needs NCS version of the imgtool,
+  # but for these we just check extra parameters.
+  string(REGEX MATCH "--sha |--pure[ $]?" imgtool_extension_required "${CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS}")
```

Comment:
why are there not Kconfigs for these? These should not be put into extra imgtool args, they should be Kconfigs which force the options